x/poolmanager/router.go

```diff
@@ -741,3 +741,93 @@ func (k Keeper) TotalLiquidity(ctx sdk.Context) (sdk.Coins, error) {
 	totalLiquidity := totalGammLiquidity.Add(totalConcentratedLiquidity...).Add(totalCosmwasmLiquidity...)
 	return totalLiquidity, nil
 }
+
+// nolint: unused
+// trackVolume converts the input token into OSMO units and adds it to the global tracked volume for the given pool ID.
+// Fails quietly if an OSMO paired pool cannot be found, although this should only happen in rare scenarios where OSMO is
+// removed as a base denom from the protorev module (which this function relies on).
+//
+// CONTRACT: `volumeGenerated` corresponds to one of the denoms in the pool
+// CONTRACT: pool with `poolId` exists
+func (k Keeper) trackVolume(ctx sdk.Context, poolId uint64, volumeGenerated sdk.Coin) {
```

Comment:
nit: thoughts on renaming this to `trackVolumeInOsmo` unless we want to expand this function more in future?

Reply:
Hmm I agree with the sentiment although I think the correct abstraction here is `trackVolume` since we want to make a single call in swap logic. I opted to not create a wrapped `trackVolume { return `trackVolumeInOsmo` } function to avoid function clutter at this stage, especially since there's a good chance we'll only ever need to track vol in OSMO

Reply:
i see, i mostly meant changing the function name to `trackVolumeInOsmo` instead of creating a wrapper function `trackVolume` that calls `trackVolumeInOsmo`